### PR TITLE
Explain WebSocket name change and remove obsolete docs part

### PR DESCRIPTION
**Pull request by ktoso**
Resolves https://github.com/akka/akka/issues/19737

### CODE REVIEW

File: akka-docs/rst/additional/http.rst [hunk 1]
```diff
@@ -1,26 +0,0 @@
...
-=========
-
-If you are using Akka Mist (Akka's old HTTP/REST module) with Akka 1.x and wish to upgrade to 2.x
-there is now a port of Akka Mist to Akka 2.x. You can find it `here <https://github.com/thenewmotion/akka-http>`_.
-
-Other Alternatives
-==================
-
-There are a bunch of other alternatives for using Akka with HTTP/REST. You can find some of them
-among the `Community Projects <http://akka.io/community>`_.
```
Comment (ktoso) [inline akka-docs/rst/additional/http.rst]: I thought we may want to remove this section having HTTP merged, right?

**Comment by johanandren** [comment]
LGTM

**Comment by 2m** [comment]
LGTM

**Comment by ktoso** [comment]
Would like one more eye on the removal of `akka-docs/rst/additional/http.rst` if we're fine with that.

File: akka-docs/rst/additional/http.rst (hunk 1 above)
Comment (johanandren) [inline akka-docs/rst/additional/http.rst]: Is it not linked from somewhere, toc. etc, that now also needs to be updated?

Comment (ktoso) [inline akka-docs/rst/additional/http.rst]: good catch, fixed

**Pull request merged**